README.md

```diff
@@ -1,25 +1,56 @@
-
-[![PyPI](https://img.shields.io/pypi/v/oauthenticator.svg)](https://pypi.python.org/pypi/oauthenticator)
-[![Build Status](https://travis-ci.org/jupyterhub/oauthenticator.svg?branch=master)](https://travis-ci.org/jupyterhub/oauthenticator)
-
-OAuth + JupyterHub Authenticator = OAuthenticator
-
-OAuthenticator currently supports the following authentication services:
-
-- [Auth0](oauthenticator/auth0.py)
-- [Azure AD](#azure-ad-setup)
-- [Azure AD B2C](#azure-ad-b2c-setup)
-- [Bitbucket](oauthenticator/bitbucket.py)
-- [CILogon](oauthenticator/cilogon.py)
-- [GitHub](#github-setup)
-- [GitLab](#gitlab-setup)
-- [Globus](#globus-setup)
-- [Google](#google-setup)
-- [MediaWiki](oauthenticator/mediawiki.py)
-- [Moodle](#moodle-setup)
-- [Okpy](#okpyauthenticator)
-- [OpenShift](#openshift-setup)
-
-A [generic implementation](oauthenticator/generic.py), which you can use with
-any provider, is also available.
+# OAuth + JupyterHub Authenticator = OAuthenticator :heart:
+
+[![Documentation build status](https://img.shields.io/readthedocs/oauthenticator?logo=read-the-docs)](https://oauthenticator.readthedocs.org/en/latest)
+[![TravisCI (.org) build status](https://img.shields.io/travis/jupyterhub/oauthenticator/master?logo=travis)](https://travis-ci.org/jupyterhub/oauthenticator)
+[![Latest PyPI version](https://img.shields.io/pypi/v/oauthenticator?logo=pypi)](https://pypi.python.org/pypi/oauthenticator)
+[![Latest conda-forge version](https://img.shields.io/conda/vn/conda-forge/oauthenticator?logo=conda-forge)](https://anaconda.org/conda-forge/oauthenticator)
+[![GitHub](https://img.shields.io/badge/issue_tracking-github-blue?logo=github)](https://github.com/jupyterhub/oauthenticator/issues)
+[![Discourse](https://img.shields.io/badge/help_forum-discourse-blue?logo=discourse)](https://discourse.jupyter.org/c/jupyterhub)
+[![Gitter](https://img.shields.io/badge/social_chat-gitter-blue?logo=gitter)](https://gitter.im/jupyterhub/jupyterhub)
+
+[OAuth](https://en.wikipedia.org/wiki/OAuth) is a token based login mechanism that doesn't rely on a username and password mapping.
+In order to use this login mechanism with JupyerHub the login handlers need to be overridden.
+OAuthenticator overrides these handlers for the common OAuth providers allowing them to be
+plugged in and used with JupyterHub.
+
+The following authentication services are supported through their own authenticator: [Auth0](oauthenticator/auth0.py),
+[Azure AD](oauthenticator/azuread.py), [Bitbucket](oauthenticator/bitbucket.py), [CILogon](oauthenticator/cilogon.py),
+[GitHub](oauthenticator/github.py), [GitLab](oauthenticator/gitlab.py), [Globus](oauthenticator/globus.py),
+[Google](oauthenticator/google.py), [MediaWiki](oauthenticator/mediawiki.py), [Okpy](oauthenticator/okpy.py),
+[OpenShift](oauthenticator/openshift.py).
+
+There is also a [GenericAuthenticator](oauthenticator/generic.py)
+that can be used to write your own Authenticators with any OAuth 2.0 provider.
```

Comment:
> can be used to write your own Authenticators with any OAuth 2.0 provider.

This sounds to me like we can only use the generic authenticator to create a custom class from it, but it can also be used on its own through configuration with any OAuth 2.0 provider. Sometimes though, we need something additional, and then a custom class is needed that can use the generic authenticator as a base class.